MantisBT: Update to v2.26.3

mreid-tt opened the pull request:
## Description

This PR contains the following:

1. Upgrade Mantisbt to version 2.26.3
2. Modernise package scripts
3. Add PHP profiles for DSM 6
4. Add full backup and restore

Fixes #4034

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relevant tags.-->
- [x] Package update

th0ma7 commented:
I believe you have it all nailed down as your explanation seems ti makes sense.  Now how to fix it?  I did a quick search and you may be stuck with this:
```
build/travis_before_script.sh:curl --data "${query_string:1}" http://$HOSTNAME:$PORT/admin/install.php
```
One way would be to create a `cross/mantisbt/patches` directory with a patch such as (I don't believe this fix is what you actually need but servers as example):
```
$ diff -uprN build/travis_before_script.sh.orig build/travis_before_script.sh
--- build/travis_before_script.sh.orig	2024-08-25 04:07:38.000000000 -0400
+++ build/travis_before_script.sh	2024-09-25 19:23:32.805340608 -0400
@@ -141,7 +141,7 @@ do
 done
 
 # trigger installation
-curl --data "${query_string:1}" http://$HOSTNAME:$PORT/admin/install.php
+curl --data "${query_string:1}" http://$HOSTNAME:$PORT/admin/install.php || true
 
 
 # -----------------------------------------------------------------------------
```
If you look at the `spksrc.patch.mk` it explain the variations of subdirectories that can be used to get trigerred for only specific arch, tcversion and al:
```
#    patches/*.patch      ## this is the only location for native and noarch builds
#    patches/kernel-$(subst +,,$(TC_KERNEL))/*.patch   ## Discard any ending +
#    patches/DSM-$(TCVERSION)/*.patch
#    patches/$(group)/*.patch
#    patches/$(group)-$(TCVERSION)/*.patch
#    patches/$(arch)/*.patch
#    patches/$(arch)-$(TCVERSION)/*.patch
```
So you could have a patch that modifies the mantisbt installer script, trigerred only under dsm7, that changes the order of things the way you actually need it to be.

With hope this gives you a few pointers to solve the issue...

BTW, looking at your install script, that`s a real piece of art :smiley:

mreid-tt commented:
Hey @th0ma7, thanks so much for your guidance! After some digging, I found a part of the installer script that was fairly easy to patch. Initial tests on DSM 7 have been successful, and I’ll be continuing with the rest of the tests shortly.

> BTW, looking at your install script, that`s a real piece of art 😃

Appreciate the kind words! It took a lot of ideation and input from several folks when I first started modernizing ownCloud a few months back.

mreid-tt commented:
@hgy59, I’ve updated another legacy package to support MariaDB and added new script features, including dynamic PHP profiles for DSM6 and a full backup and restore function. Looking forward to your review and approval.
